client/src/screens/events/Detail.js

```diff
@@ -92,48 +138,98 @@ class Detail extends Component {
   }
 
   onRefresh = () => {
-    // NYI
+    // NYI very well
     this.props.refetch();
   };
 
-  mapOnLayout = (marks, locations) => {
-    const mergedpoints = marks.concat(locations);
+androidLocationPermission = async () => {
+  console.log('Checking android permissions');
+  const granted = await PermissionsAndroid.request(
+    PermissionsAndroid.PERMISSIONS.ACCESS_FINE_LOCATION,
+    {
+      title: 'Permission Needed',
+      message: 'We need permission to track your location accuratly.',
+    },
+  );
+  console.log('permissions:', granted);
+  return granted === PermissionsAndroid.RESULTS.GRANTED;
+}
+
+mapZoomMe = () => {
+  if (this.state.myPosition) {
+    const { longitude, latitude } = this.state.myPosition;
+    const myLocation = [{
+      longitude,
+      latitude,
+    }];
+    this.zoomMap(myLocation);
+  }
+}
+
+  mapOnLayout = () => {
+    let mergedpoints = [];
+    if (this.state.myPosition) {
+      const { longitude, latitude } = this.state.myPosition;
+      const myLocation = [{
+        longitude,
+        latitude,
+      }];
+      mergedpoints = myLocation.concat(this.state.eventMarkers);
+    } else {
+      mergedpoints = this.state.eventMarkers;
+    }
+    // Zoom to the user and events if possible, otherwise start around events
+
+    this.zoomMap(mergedpoints);
+  }
+
+  zoomMap(onThis) {
```

Comment:
maybe rather than `onThis` which could be construed as an event handler use `coords`?

Reply:
how about fitToTheseCoordinates?